### PR TITLE
fix: use metadataProvider option instead of cornerstone.metaData

### PR DESCRIPTION
**Pull request by Punzo**

### CODE REVIEW

**Review by pieper — APPROVED** [review]
LGTM.

**Comment by pieper** [comment]
@Punzo do you know why the test check is failing?  Also I had thought @swederik turned off circleci - but it looks like we are still expecting it here?

**Comment by Punzo** [comment]
> @Punzo do you know why the test check is failing? Also I had thought @swederik turned off circleci - but it looks like we are still expecting it here?

not sure, locally the tests are all green. It looks like the npm install step failed:
image

is it possible for you to rerun it? I don't have the permissions for it.

**Comment by pieper** [comment]
Thank you Davide, yes, it must have been a glitch since it worked the second time.

I found these notes and have turned off circleci check requirement before merging.  
https://circleci.com/docs2/2.0/enable-checks

Hopefully it will take effect for future PRs but for now I'll bypass the check and merge.

File: src/adapters/Cornerstone/Segmentation_4X.js [hunk 1]
```diff
@@ -1105,10 +1103,7 @@ function insertPixelDataPlanar(
             continue;
         }
 
-        const sourceImageMetadata = cornerstone.metaData.get(
-            "instance",
-            imageId
-        );
+        const sourceImageMetadata = metadataProvider.get("instance", imageId);
```
Comment (rogerbramon) [inline src/adapters/Cornerstone/Segmentation_4X.js:1106]: Thanks @Punzo for applying the changes so fast! I have doubt, though. Who is adding support for the "instance" type?

Comment (Punzo) [reply src/adapters/Cornerstone/Segmentation_4X.js:1106]: > Thanks @Punzo for applying the changes so fast! 

you are welcome,

> I have doubt, though. Who is adding support for the "instance" type?

 not sure I have understood what do you mean. the cornerstone provider does provide all the metadata of the instance when using metadataProvider.get("instance", imageId);

Comment (rogerbramon) [reply src/adapters/Cornerstone/Segmentation_4X.js:1106]: When trying it out, `sourceImageMetadata` is undefined. However, if I comment out the ROWS-COLUMNS check, the segmentation loads correctly. 

I debugged the `metadaProvider.get` method, and it seems there's no registered provider that responds to the `instance` type. Thus, I was wondering if I miss any initialization that adds a provider to have the instance type available. 

Thanks

Comment (rogerbramon) [reply src/adapters/Cornerstone/Segmentation_4X.js:1106]: Hey @Punzo, I would appreciate if you could help me on this. Thanks!

Comment (Punzo) [reply src/adapters/Cornerstone/Segmentation_4X.js:1106]: Hi, 

you can check 

https://github.com/OHIF/Viewers/blob/master/extensions/cornerstone/src/init.js for cornerstone initialization.

here is where the segmentation is parsed in OHIF using dcmjs:
https://github.com/OHIF/Viewers/blob/c554c5fac2dfb8cf4800ceffd936475e07ca7782/extensions/dicom-segmentation/src/getOHIFDicomSegSopClassHandler.js#L152

and 

Can you also move thiss discussion to an issue in the issue tracker, please?

**Pull request merged**